eng/pipelines/templates/steps/publish-cli-winget.yml

```diff
@@ -0,0 +1,53 @@
+parameters:
+  - name: AllowPrerelease
+    type: boolean
+    default: true
+    displayName: Allow prerelease versions to be published. Should be "false" after GA.
+  - name: CLiVersion
+    type: string
+    default: $(CLI_VERSION)
+    displayName: CLI version
+  - name: GitHubToken
+    type: string
+    default: $(azuresdk-github-pat)
+    displayName: GitHub token
+
+steps:
+  - task: PowerShell@2
+    displayName: Set WinGet Version
+    inputs:
+      pwsh: true
+      targetType: filePath
+      filePath: eng/scripts/Get-WinGetVersion.ps1
+      arguments: >-
+        -Version '${{ parameters.CliVersion }}'
+        -DevOpsOutput
+
+  - pwsh: |
+      $submitPackage = eng/scripts/Test-ShouldReleaseWinGet.ps1 `
+        -Version '$(WinGetVersion)' `
+        -AllowPrerelease:$${{ parameters.AllowPrerelease }}
+
+      if ('$(Skip.ReleaseWinGet)' -eq 'true') {
+        $submitPackage = 'false'
+      }
+      Write-Host "Setting SubmitWinGetPackage to $submitPackage"
+      Write-Host "##vso[task.setvariable variable=SubmitWinGetPackage]$submitPackage"
+    displayName: Set SubmitWinGetPackage
+
+  - task: PowerShell@2
```

Comment:
Bit of a nit, really, but why keep switching between both forms? You can run a file and pass args with the `pwsh:` form, which is more succinct. You do that on line 27, for example.

Reply:
Using the `PowerShell@2` form can help (slightly) in debugging -- https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/2/Azure-DevOps-Pipeline-Guidance?anchor=devops-tasks-vs-pwsh-script-shortcut